### PR TITLE
Fix arg confusion

### PR DESCRIPTION
**Pull request by HParker**
`get('/posts', params: { filters: { body2: ["hi", "hello"] } })`
was being used for the scope,
```ruby
scope :body2, -> (text) {
  where(body: text)
}
```

but,
`collection = Filterable::Filtrator.filter(Post.all, { filters: { body_and_priority: ["foo", 10] } }, [filter])`
with,
```ruby
scope :body_and_priority, -> (text, priority) {
  where(body: text, priority: priority)
}
```

One is wrong, so I changed the top scope to,
```ruby
scope :body2, -> (*text) {
  where(body: text)
}
```

In the test.

### CODE REVIEW

**Comment by HParker** [comment]
Do we need a doc for this? It feels weird explaining it.

"If you want to have your scope take an array of args, you have to specify that you want all the args passed to it."

File: lib/filterable/filter.rb [hunk 1]
```diff
@@ -53,6 +45,18 @@ def type_validator
 
     private
 
+    def apply_scope_filter(collection, value, active_sorts_hash, params)
+      if value.present? && scope_params.empty?
+        collection.public_send(internal_name, *parameter(value))
```
Comment (Oneill38) [inline lib/filterable/filter.rb]: The `*parameter(value)` breaks a bunch of procore filters & tests I believe

Comment (HParker) [reply lib/filterable/filter.rb]: I suspect those scopes on the Procore side need to change. I am not sure how we can support both behaviors.

**Comment by HParker** [comment]
Ok, I think I was confused here. the weird usage is actually having multi arg scopes. I have no idea how a client should use that. Pass an array where the order of the elements matters?

That doesn't make a lot of sense to me. Let me think about this more.

**Pull request closed without merging**